docs/src/further_topics/ugrid/data_model.rst

```diff
@@ -0,0 +1,326 @@
+    * | Optional: :attr:`~iris.experimental.ugrid.Mesh.edge_coords`,
+        :attr:`~iris.experimental.ugrid.Mesh.face_coords`
+      | For indicating the 'centres' of the edges/faces.
+
+  * 1 or more :class:`iris.experimental.ugrid.Connectivity`\s:
+
+    * | **Required for 1D (edge) elements**:
+        :attr:`~iris.experimental.ugrid.Mesh.edge_node_connectivity`
+      | Define the edges by connecting nodes.
+    * | **Required for 2D (face) elements**:
+        :attr:`~iris.experimental.ugrid.Mesh.face_node_connectivity`
+      | Define the faces by connecting nodes.
+    * Optional: any other connectivity type. See
+      :attr:`iris.experimental.ugrid.mesh.Connectivity.UGRID_CF_ROLES` for the
+      full list of types.
+
+.. todo: MeshCoord printout?
+
+* | :class:`iris.experimental.ugrid.MeshCoord`
+  | Described in detail in `MeshCoords`_.
+  | Stores the following information:
+
+    * | :attr:`~iris.experimental.ugrid.MeshCoord.mesh`
+      | The :class:`~iris.experimental.ugrid.Mesh` associated with this
+        :class:`~iris.experimental.ugrid.MeshCoord`. Mirrored by the
+        :attr:`~iris.cube.Cube.mesh` attribute of any :class:`~iris.cube.Cube`
+        this :class:`~iris.experimental.ugrid.MeshCoord` is attached to (see
+        `The Basics`_)
+
+    * | :attr:`~iris.experimental.ugrid.MeshCoord.location`
+      | ``node``/``edge``/``face`` - the element detailed by this
+        :class:`~iris.experimental.ugrid.MeshCoord`. Mirrored by the
+        :attr:`~iris.cube.Cube.location` attribute of any
+        :class:`~iris.cube.Cube` this
+        :class:`~iris.experimental.ugrid.MeshCoord` is attached to (see
+        `The Basics`_).
+
+MeshCoords
+~~~~~~~~~~
+Links a :class:`~iris.cube.Cube` to a :class:`~iris.experimental.ugrid.Mesh` by
+attaching to the :class:`~iris.cube.Cube`\'s unstructured dimension, in the
+same way that all :class:`~iris.coords.Coord`\s attach to
+:class:`~iris.cube.Cube` dimensions. This allows a single
+:class:`~iris.cube.Cube` to have a combination of unstructured and structured
+dimensions (e.g. horizontal mesh plus vertical levels and a time series),
+using the same logic for every dimension.
+
+:class:`~iris.experimental.ugrid.MeshCoord`\s are instantiated using a given
+:class:`~iris.experimental.ugrid.Mesh`, ``location``
+("node"/"edge"/"face") and ``axis``. The process interprets the
+:class:`~iris.experimental.ugrid.Mesh`\'s
+:attr:`~iris.experimental.ugrid.Mesh.node_coords` and if appropriate the
+:attr:`~iris.experimental.ugrid.Mesh.edge_node_connectivity`/
+:attr:`~iris.experimental.ugrid.Mesh.face_node_connectivity` and
+:attr:`~iris.experimental.ugrid.Mesh.edge_coords`/
+:attr:`~iris.experimental.ugrid.Mesh.face_coords`
+to produce a :class:`~iris.coords.Coord`
+:attr:`~iris.coords.Coord.points` and :attr:`~iris.coords.Coord.bounds`
+representation of all the :class:`~iris.experimental.ugrid.Mesh`\'s
+nodes/edges/faces for the given axis.
+
+A :class:`~iris.experimental.ugrid.Mesh`
```

Comment:
Reads a bit awkwardly.  I think "The method :meth:`~iris.experimental.ugrid.Mesh.to_MeshCoords` is available ... *given* only the ``location`` argument."

Reply:
a0b405c3b4432e6e172616620cb6a1563b083a57